File: tests/bindings/vbnet/TestBrickletDualRelay.vb
Imports Tinkerforge

Module TestBrickletDualRelay
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "xyz" ' Change to your UID

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim dr As New BrickletDualRelay(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Turn both relays off and on
        dr.SetState(False, False)
        System.Threading.Thread.Sleep(1000)
        dr.SetState(True, True)

        ipcon.Disconnect()
    End Sub
End Module

File: tests/bindings/octave/test_bricklet_dual_relay.m
function test_bricklet_dual_relay()
    more off;
    
    HOST = "localhost";
    PORT = 4223;
    UID = "xyz"; % Change to your UID

    ipcon = java_new("com.tinkerforge.IPConnection"); % Create IP connection
    dr = java_new("com.tinkerforge.BrickletDualRelay", UID, ipcon); % Create device object

    ipcon.connect(HOST, PORT); % Connect to brickd
    % Don"t use device before ipcon is connected

    % Turn both relays off and on
    dr.setState(false, false);
    pause(1);
    dr.setState(true, true);

    ipcon.disconnect();
end

test_bricklet_dual_relay();
